src/Runner.Worker/JobRunner.cs

```diff
@@ -206,6 +214,93 @@ public async Task<TaskResult> RunAsync(Pipelines.AgentJobRequestMessage message,
             }
         }
 
+        private bool JobPassesSecurityRestrictions(IExecutionContext jobContext)
+        {
+            var gitHubContext = jobContext.ExpressionValues["github"] as GitHubContext;
+
+            try {
```

Comment:
I'm not familiar with the `runner` codebase, but besides from a few `as` casts that could result in a `null`-reference, I don't see anything in the `OkayToRunPullRequest()` method that could potentially throw an exception.

You might rather want to remove that `try..catch` block and always return `false` in `OkayToRunPullRequest()` if one of the casts results in a `null` value.